.github/workflows/scanner-build.yaml

```diff
@@ -30,26 +36,38 @@ jobs:
         source './scripts/ci/lib.sh'
 
         # If goarch is updated, be sure to update architectures in "push-scanner-manifests" below.
-        matrix='{ "build_and_push": { "name":["default"], "goos":["linux"], "goarch":["amd64", "arm64"] }, "push_manifests": { "name":["default"] } }'
+        matrix='{
+          "pre_build_scanner_go_binary": { "name":["default"], "goos":["linux"], "goarch":["amd64", "arm64"] },
+          "build_and_push_scanner": { },
+          "push_scanner_manifests": { "name":["default"], "registry":[] },
+          "scan_images_with_roxctl": { "image":["scanner-v4", "scanner-v4-db"], "registry":[] }
+        }'
 
         if ! is_in_PR_context || pr_has_label ci-build-all-arch; then
-          matrix="$(jq '.build_and_push.goarch += ["ppc64le", "s390x"]' <<< "$matrix")"
+          matrix="$(jq '.pre_build_scanner_go_binary.goarch += ["ppc64le", "s390x"]' <<< "$matrix")"
         fi;
 
         # Conditionally add a prerelease build (binaries built with GOTAGS=release)
         if ! is_tagged; then
           if ! is_in_PR_context || pr_has_label ci-build-prerelease; then
-            matrix="$(jq '.build_and_push.name += ["prerelease"]' <<< "$matrix")"
-            matrix="$(jq '.push_manifests.name += ["prerelease"]' <<< "$matrix")"
+            matrix="$(jq '.pre_build_scanner_go_binary.name += ["prerelease"]' <<< "$matrix")"
+            matrix="$(jq '.push_scanner_manifests.name += ["prerelease"]' <<< "$matrix")"
           fi
         fi
 
         # Conditionally add a -race debug build (binaries built with -race)
         if ! is_in_PR_context || pr_has_label ci-build-race-condition-debug; then
-          matrix="$(jq '.build_and_push.name += ["race-condition-debug"]' <<< "$matrix")"
-          matrix="$(jq '.push_manifests.name += ["race-condition-debug"]' <<< "$matrix")"
+          matrix="$(jq '.pre_build_scanner_go_binary.name += ["race-condition-debug"]' <<< "$matrix")"
+          matrix="$(jq '.push_scanner_manifests.name += ["race-condition-debug"]' <<< "$matrix")"
         fi
 
+        matrix="$(jq '.build_and_push_scanner = .pre_build_scanner_go_binary' <<< "$matrix")"
```

Comment:
Since they'll always be the same, is there a purpose of splitting this up into two variables?

Reply:
The matrix of `build_and_push_scanner` is the one of `pre_build_scanner_go_binary` **plus** the `registry` (e.g. see the output [here](https://github.com/stackrox/stackrox/actions/runs/12637604376/job/35212133989#step:3:61)).
I hope with the recent change the code makes it even more clear.
The reason I couldn't declare attributes on `build_and_push_scanner` is because `pre_build_scanner_go_binary` gets constructed dynamically based on the labels and we need the same content in `build_and_push_scanner`. The copying isn't elegant but it's best I could come up with.

Reply:
ah ok I missed the next line where you add the registry. I see how they are different now